### DISCUSSION TITLE
Problems with SMS 2FA ?

### DISCUSSION THREAD

**Discussion by Amapola-Negra** [community/community#129638] [New to GitHub] [answered]
Hi everybody. I have entered in my account today and I have the following messagge: " Please configure another 2FA method to reduce your risk of permanent account lockout. We strongly recommend against SMS as it is prone to fraud and delivery may be unreliable depending on your region." I don't know what have changed since I configured the SMS. My country is amongst thouse in which SMS are alowed according to GitHub list. So I don't understand why my account could be blocked. First of all I want to know if this is just a recommendation or an obligation, I have an app in my mobile to generate 2FA is there any way I could use it for GitHub? I have seen as well there is a way to use a security Key as a 2FA but I don't know how to. Thanks in advance, I would appreciate any help with this.

Regards

**Comment by davevad93** [comment]
Hi @Amapola-Negra ,

The warning is a bit confusing, it basically indicates that it's recommended to use more than one 2FA method and avoid using SMS authentication for security reasons. I've added authentication via `Mobile App` and the warning no longer pops up.

For further reference, there was a discussion about this a few days ago. #129189

**Reply by Amapola-Negra** [reply]
Well. It appeards to highlight the same concern regarding the account blocking or cancel, and no answer. Any clue? 

I want to know ho to use security keys with 2FA. Thanks again, :)

**Reply by davevad93** [reply, marked as answer]
@Amapola-Negra I think that the following warning "***Please configure another 2FA method to reduce the risk of permanent account lockout***" concerns adding an additional 2FA method to reduce the risk of losing your account in case of hacking and avoiding SMS authentication because isn't very safe, (I've seen many people here in the last few months claiming that their account was hacked). I highly doubt that means `GitHub` will directly lock an account if the above conditions aren't met, it's just an informative warning, but that's only my impression, to be 100% sure we would need a confirmation from someone from the staff. About using security keys with 2FA sorry but I'm currently not using that as a 2FA method, I'm using the `Authenticator app` as the main authentication method and it works fine. You can activate several 2FA methods and choose the one you prefer on the main login page.

**Reply by biofsphere** [reply]
I did try to configure another 2fa method by choosing SMS Message on my mobile phone. However, I was unable to register my phone number as the international code for Brazil is missing from the countries list provided by GitHub.

**Reply by maxpopovitch** [reply]
My country (Ukraine) is not in the list of countries to setup another 2FA via SMS/Text message. Could you please add it?

**Reply by wmport** [reply]
> My country (Ukraine) is not in the list of countries to setup another 2FA via SMS/Text message. Could you please add it?

Good 👍. 
long answer github  ((

**Reply by Hi-Angel** [reply]
@davevad93 the discussion you linked doesn't seem to have any details on why SMSes are prone to fraud though. Per my reading of the OP's title, they seem to be wondering about that too: what's wrong with SMS.

I think Github ought to write an article *(or link to an existing one)* about that before putting on a warning for unclear reasons.

**Reply by davevad93** [reply]
Hey @Hi-Angel a couple of weeks ago, someone from the GitHub staff has given an official answer about this 2FA thing. Here's the link: https://github.com/orgs/community/discussions/129244#discussioncomment-10024978

**Reply by Hi-Angel** [reply]
But the link doesn't seem to explain what's wrong with SMS, does it?

**Reply by davevad93** [reply]
From what I understood, their advice is for users who still use SMS authentication, which isn't a secure method because for instance if you lose your phone you could lose access to your account for good; so they are encouraging users to enable multiple 2FA methods and avoid SMS authentication; I personally don't use it for that reason alone. Now, I agree with you with the fact that there was a lack of clarity from GitHub regarding the warning and that they should provide more details, but essentially I think the reason is because it's considered the least secure method.

**Reply by Hi-Angel** [reply]
> if you lose your phone you could lose access to your account for good

Huh, can't you just make a new SIM-card? Idk if it's different in other countries, but in my country every SIM-card is bound to passport data. So even if you lose your SIM-card, you can always just make a new one.

**Reply by Hi-Angel** [reply]
…rather, in my country a phone number is bound to passport data, whereas SIM-card is just a way to make calls and stuff with the phone number. The point stands though: if I lose my phone, I can just go order a new SIM card.

**Reply by davevad93** [reply]
That's true @Hi-Angel , it would makes more sense that they want users to drop SMS 2FA because SMS aren't encrypted and are vulnerable to various types of attacks. Maybe they've noticed that some users have been hacked using this kind of authentication? I really don't know, mine are just suppositions to be fair.
